app/helpers/concerns/foreman_remote_execution/hosts_helper_extensions.rb

```diff
@@ -11,9 +11,13 @@ def multiple_actions_with_remote_execution
       multiple_actions_without_remote_execution + [[_('Run Job'), new_job_invocation_path, false]]
     end
 
-    def host_title_actions_with_run_button(*args)
-      title_actions(button_group(link_to(_('Run Job'), new_job_invocation_path(:host_ids => [args.first.id]), :id => :run_button)))
-      host_title_actions_without_run_button(*args)
+    def host_title_actions_with_run_button(host)
+      links = [link_to(_('Run Job'), new_job_invocation_path(:host_ids => [host.id]), :id => :run_button)]
+      if RemoteExecutionFeature.feature(:reprovision).template && @host.provisioning_template(:kind => 'script')
+        links << link_to(_('Reprovision'), reprovision_host_path(host.id), { :method => :post, :disabled =>  @host.build })
```

Comment:
Could we somehow add this to the build button as a dropdown? I would guess maybe not easy to extend from plugin.

I only wonder because  the host actions can get quite big if you install a lot of plugins, and this is really related to the build.